resources/osl_gitlfs.rb

```diff
@@ -0,0 +1,21 @@
+resource_name :osl_gitlfs
+
+default_action :sync
+
+provides :osl_gitlfs
+property :directory, String,  required: true
+property :repository, String, required: true
+
+action :sync do
+  run_context.include_recipe 'base::gitlfs'
+  run_context.include_recipe 'git'
+
+  git new_resource.directory do
+    repository new_resource.repository
+    environment('GIT_LFS_SKIP_SMUDGE' => '1')
+  end
+
+  execute 'git lfs pull' do
+    cwd "#{new_resource.directory}/#{new_resource.name}"
+  end
```

Comment:
The resource should have ``#{new_resource.name}`` included so that it's not cloned when using it multiple times.
```suggestion
  execute "git lfs pull #{new_resource.name}" do
    command 'git lfs pull'
    cwd "#{new_resource.directory}/#{new_resource.name}"
  end
```